### PR TITLE
Extract behavior

### PR DESCRIPTION
**Pull request by rogeralsing**
This PR extracts the become/unbecome functionality into a separate object.

### CODE REVIEW

File: src/Proto.Actor/Behavior.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+//       Copyright (C) 2015-2017 Asynkron HB All rights reserved
+//   </copyright>
+// -----------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Proto
+{
+    public class Behavior
```
Comment (rogeralsing) [inline src/Proto.Actor/Behavior.cs]: The new behavior class

File: src/Proto.Actor/Behavior.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            Become(receive);
+        }
+
+        public void Become(Receive receive)
+        {
+            _behaviors.Clear();
+            _behaviors.Push(receive);
+        }
+
+        public void BecomeStacked(Receive receive)
```
Comment (rogeralsing) [inline src/Proto.Actor/Behavior.cs]: Has both Become and BecomeStacked methods

File: tests/Proto.Actor.Tests/PopPushBehaviorTests.cs [hunk 3]
```diff
@@ -13,21 +10,23 @@ public class PopPushBehaviorTests
         [Fact]
         public async Task PopBehaviorShouldRestorePushedBehavior()
         {
-            PID pid = SpawnActorFromFunc(ctx =>
+            var behavior = new Behavior();
+            behavior.Become(ctx =>
```
Comment (rogeralsing) [inline tests/Proto.Actor.Tests/PopPushBehaviorTests.cs]: Updated test to use the new Behavior class

**Comment by tomliversidge** [comment]
Looks good to me!

**Pull request merged**